### PR TITLE
Add docker actions

### PR DESCRIPTION
**Pull request by jackshirazi**
I tried to test this, the docker image built but it failed on attestation (maybe because push was false in testing?)

### CODE REVIEW

**Comment by v1v** [comment]
> maybe because push was false in testing?)

maybe `push-to-registry: true` was set?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -128,6 +129,65 @@ jobs:
...
 
+  build-and-push-docker-images:
+    name: "Build and push docker images"
+    runs-on: ubuntu-latest
+    needs:
+      - await-maven-central-artifact
+      - create-github-release
+    env:
+      DOCKER_IMAGE_NAME: docker.elastic.co/observability/elastic-otel-javaagent
+    steps:
```
Comment (v1v) [inline .github/workflows/release.yml:140]:
```suggestion
    permissions:
      attestations: write
      contents: read
      id-token: write
    steps:
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -128,6 +129,65 @@ jobs:
...
+          file: "docker/Dockerfile"
+          push: true
+          tags: ${{ steps.docker-meta.outputs.tags }}
+          labels: ${{ steps.docker-meta.outputs.labels }}
+          build-args: |
+            JAR_FILE=elastic-otel-javaagent.jar
+            EXTENSION_JAR_FILE=elastic-otel-agentextension.jar
+
+      - name: generate build provenance (containers)
+        uses: github-early-access/generate-build-provenance@main
```
Comment (v1v) [inline .github/workflows/release.yml:185]:
```suggestion
        actions/attest-build-provenance@173725a1209d09b31f9d30a3890cf2757ebbff0d  # v1.1.2
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -128,6 +129,65 @@ jobs:
...
+      - name: Get latest tag
+        run: echo "LATEST_TAG=$(git tag --list --sort=version:refname "v*" | grep -v RC | sed s/^v// | tail -n 1)" >> "${GITHUB_ENV}"
+
+      - uses: elastic/apm-pipeline-library/.github/actions/docker-login@current
+        with:
+          registry: docker.elastic.co
+          secret: secret/apm-team/ci/docker-registry/prod
+          url: ${{ secrets.VAULT_ADDR }}
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
```
Comment (v1v) [inline .github/workflows/release.yml:155]: If possible, let's remove this dependency with Vault and use GitHub secrets. I'll prepare those secrets and then suggest the change here

Comment (v1v) [reply .github/workflows/release.yml:155]:
```suggestion
      - name: Log in to the Elastic Container registry
        uses: docker/login-action@e92390c5fb421da1463c202d546fed0ec5c39f20 # v3.1.0
        with:
          registry: ${{ secrets.ELASTIC_DOCKER_REGISTRY }}
          username: ${{ secrets.ELASTIC_DOCKER_USERNAME }}
          password: ${{ secrets.ELASTIC_DOCKER_PASSWORD }}
```

**Comment by jackshirazi** [comment]
> maybe `push-to-registry: true` was set?

it was, you're right

**Comment by v1v** [comment]
> it was, you're right

I think the lack of specific permissions in the job where the attest happened were the reason :)

**Review by v1v — APPROVED** [review]

**Pull request merged**